public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -517,15 +534,33 @@ export class LokiDatasource
   }
 
   getLogRowContext = (row: LogRowModel, options?: RowContextOptions): Promise<{ data: DataFrame[] }> => {
-    const target = this.prepareLogRowContextQueryTarget(
-      row,
-      (options && options.limit) || 10,
-      (options && options.direction) || 'BACKWARD'
-    );
+    const direction = (options && options.direction) || 'BACKWARD';
```

Comment:
I like that you extracted direction into variable. I am thinking that it would be a good idea to do the same for limit.